Source/DafnyCore/Backends/Dafny/AST.dfy

```diff
@@ -19,13 +38,22 @@ module {:extern "DAST"} DAST {
     Set(element: Type) |
     Multiset(element: Type) |
     Map(key: Type, value: Type) |
+    SetBuilder(element: Type) |
+    MapBuilder(key: Type, value: Type) |
     Arrow(args: seq<Type>, result: Type) |
     Primitive(Primitive) | Passthrough(string) |
     TypeArg(Ident)
 
   datatype Primitive = Int | Real | String | Bool | Char
 
-  datatype ResolvedType = Datatype(path: seq<Ident>) | Trait(path: seq<Ident>) | Newtype(Type)
+  datatype NewtypeRange =
```

Comment:
Consider naming something like `IntegerNewtypeRange` since we have newtypes on other base types now

Reply:
As per the previous comment, I'm not renaming this to IntegerNewtypeRange because it accounts for other possible ranges in the future.